test/ng/filter/limitToSpec.js

```diff
@@ -153,19 +201,38 @@ describe('Filter: limitTo', function() {
     expect(limitTo(items, '-4', -12)).toEqual(['e', 'f', 'g', 'h']);
     expect(limitTo(str, 4, '-12')).toEqual("tuvw");
     expect(limitTo(str, '-4', -12)).toEqual("wxyz");
+    expect(limitTo(arrayLike, 4, '-12')).toEqual(['a', 'b', 'c', 'd']);
+    expect(limitTo(arrayLike, '-4', -12)).toEqual(['e', 'f', 'g', 'h']);
   });
 
   it('should return the entire string beginning from Y if X is positive and X+Y exceeds input length', function() {
     expect(limitTo(items, 7, 3)).toEqual(['d', 'e', 'f', 'g', 'h']);
     expect(limitTo(items, 7, -3)).toEqual(['f', 'g', 'h']);
     expect(limitTo(str, 6, 3)).toEqual("wxyz");
     expect(limitTo(str, 6, -3)).toEqual("xyz");
+    expect(limitTo(arrayLike, 7, 3)).toEqual(['d', 'e', 'f', 'g', 'h']);
+    expect(limitTo(arrayLike, 7, -3)).toEqual(['f', 'g', 'h']);
   });
 
   it('should return the entire string until index Y if X is negative and X+Y exceeds input length', function() {
     expect(limitTo(items, -7, 3)).toEqual(['a', 'b', 'c']);
     expect(limitTo(items, -7, -3)).toEqual(['a', 'b', 'c', 'd', 'e']);
     expect(limitTo(str, -6, 3)).toEqual("tuv");
     expect(limitTo(str, -6, -3)).toEqual("tuvw");
+    expect(limitTo(arrayLike, -7, 3)).toEqual(['a', 'b', 'c']);
+    expect(limitTo(arrayLike, -7, -3)).toEqual(['a', 'b', 'c', 'd', 'e']);
+  });
+
+  it('should not throw an error if used with an array like object', function() {
+    function getArguments() {
+      return arguments;
+    }
+    var argsObj = getArguments({name: 'Misko'}, {name: 'Igor'}, {name: 'Brad'});
+
+    var nodeList = jqLite("<p><span>Misko</span><span>Igor</span><span>Brad</span></p>")[0].childNodes;
+
+    expect(limitTo(argsObj, 2).length).toBe(2);
+    expect(limitTo('abc', 1).length).toBe(1);
+    expect(limitTo(nodeList, 3).length).toBe(3);
```

Comment:
This is not testing anything. You need a `limit` that is less than the actual length.